File: frontend/src/types/Restaurant.ts
export interface Restaurant {
    id: number;
    name: string;
    cuisineType: string;
    address: string;
    rating: number;
    imageUrl: string;
    distance?: number;
}

export interface RestaurantFormData {
    name: string;
    cuisineType: string;
    address: string;
    rating: number;
    imageUrl: string;
}

File: backend/src/main/java/com/foodExplorer/demo/model/Restaurant.java
package com.foodExplorer.demo.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Restaurant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String cuisineType;
    private String address;
    private Double latitude;
    private Double longitude;
    private Double rating;
    private String imageUrl;

}
